TopMovies/MoviesConnector/CategoryVCConnector.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  CategoryVCConnector.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 17.02.2021.
+//
+
+import Foundation
+
+class CategoryVCConnector: BaseConnector<MoviesCategoryVCProps> {
+  var categoryId: MovieCategory.ID?
+  
+  required init(updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    super.init(updateProps: updateProps)
+  }
+  convenience init(categoryId: MovieCategory.ID, updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    self.init(updateProps: updateProps)
+    self.categoryId = categoryId
+    mainStore.dispatch(RequestedMoviesListAction(categoryId: categoryId,
+                                                 requestType: .loadMore))
+  }
+  
+  override func newState(state: MainState) {
+    guard
+      let categoryId = categoryId,
+      let categoryName = state.movieCategoriesState.relational[categoryId]?.title,
+      let categoryState = state.categoriesPaginationState.paginated[categoryId]
+    else { return }
+    
+    if categoryState.reload.isDownloading || categoryState.loadMore.isDownloading {
+      let props = MoviesCategoryVCProps(categoryId: categoryId,
+                                        categoryName: categoryName,
+                                        isReloadInProgress: categoryState.reload.isDownloading,
+                                        isLoadMoreInProgress: categoryState.loadMore.isDownloading,
+                                        hasNewMovies: false,
+                                        movies: [])
```

Comment:
Why your list is empty during reload? What if you open category, go back and open again? IMHO you can show old list without a doubt since new info may not differ from your old one.